### DISCUSSION TITLE
Minor binary updates to busybox and other packages on the shell

### DISCUSSION THREAD

**Discussion by sideup66** [gtxaspec/wz_mini_hacks#777] [Development]
Hey guys,
I know development here is rather slow compared to thingino, but im curious if there is any plan possibly to update some of the binaries in the current build of the hack. I know we have busybox, wireguard, and a few others. I know busybox had a couple recent updates that closed some security vulns I believe for instance, not sure on busybox and others....is there possibly a plan to update some of the binaries on wz_mini hacks? By what I gather, these chips are Mips based, so im not sure where you would get binaries for these assuming precompiled? Be a good thing to discuss either way.

**Comment by endertable** [comment]
I think from a developer's standpoint, the creator maybe wants to keep on the stance 'if it ain't broke, don't fix it'.  It's very easy to update things that aren't necessary and end up breaking something else.
On the other hand, if you are interested in premade mips packages, someone posted on how to install opkg which is a package manager, that would let you install 10's of thousands of different packages for mips alone.  Mind you, it would create it's own structure for the libraries, binaries, etc and you would have to update your PATH, but it could give you other programs you are seeking.

**Comment by Bob-AOL** [comment]
I'm still waiting for ONVIF...., (which would create a whole new universe for these cams on various VMS platforms, not the least of which is Blue Iris.)

**Reply by gtxaspec** [reply]
we just got onvif working in thingino... need to investigate if it can work with wz_mini

**Comment by sideup66** [comment]
@endertable thats a fair point. However, I am not looking for EVERYTHING truthfully, nor every update. Im thinking more in the interest of just basic security vulns. For instance, I know drop bear had a couple vulns closed on a newer 2024 build of it recently, and im sure other packages like wiregaurd have seen some minor updates too. Not that its a HUGE deal in my setup (mine is behind a firewall and not exposed to wan) but it is on the local network and a bad script could possibly try to do something. I am pretty sure these are MIPS based cpus....hence the ask about precompiled binaries? if they exist, would it be as simple as one swapping out the binary in the git tree.... @gtxaspec  would likely know best there, if so id be happy to send in a pull request possibly with what I'd like to see updated.

**Comment by gtxaspec** [comment]
I think the best way would be to create a build system based on buildroot... in a similar fashion to how we build [thingino-firmware](https://github.com/themactep/thingino-firmware)  

This way, we can compile everything dynamically using CI on github, or someone can download the repo on their machine and compile everything fresh every time, and have it be updated on the fly.
